janus_server/src/datastore.rs

```diff
@@ -1187,22 +1194,22 @@ impl Transaction<'_> {
         let stmt = self
             .tx
             .prepare_cached(
-                "WITH ranges AS (
+                &format!("WITH ranges AS (
                     SELECT tsrange(x.range_start, x.range_end) as interval
                     FROM unnest($1::TIMESTAMP[], $2::TIMESTAMP[]) AS x(range_start, range_end)
                 )
                 SELECT
-                    COUNT(aggregate_share_jobs.batch_interval_start) as overlap_count,
+                    COUNT({table}.batch_interval_start) as overlap_count,
```

Comment:
Is interpolating the table name into the query too clever? I think the Rigorous, Sensible thing to do might be to normalize the schema: replace `collect_jobs` and `aggregate_share_jobs` with a single table that has the common columns, and then introduce a new table that would have the `helper_aggregate_share`, `leader_aggregate_share`, `report_count` and `checksum` columns from `collect_jobs`, plus an FK to `collect_jobs`. The advantage is that the new table's columns could all be `NOT NULL`. But I'm not sure if this kind of normalization is a good idea yet.

Reply:
I like the idea of a single, shared table between leader & helper (as long as this doesn't introduce a bunch of complexity/confusion, but from your description it sounds like it would be fairly straightforward); we get away with something similar with the `aggregation_jobs` & `report_aggregations` tables.

That said, I'm OK with interpolating the table name if you want to keep it this way and/or address in a later PR, but doing so is definitely a sign that our schema could be improved IMO.

(if you do decide to modify the schema, please let me take another look for review)

Reply:
Looking at it again, we'd also have to do something about the `collect_job_id` column for the leader. I think I'm going to consider making that change in the future, because right now, I want to keep making progress on a collect implementation we can do interop testing with. #156